From 8cc927e94999c81e2e663412782571572dfec772 Mon Sep 17 00:00:00 2001
From: Roy Pledge <Roy.Pledge@freescale.com>
Date: Tue, 2 Feb 2016 06:41:01 -0500
Subject: [PATCH 080/146] fsl_qbman: Use do_div() when dividing

The do_div() function must be used for division inside the kernel.

Signed-off-by: Roy Pledge <Roy.Pledge@freescale.com>
---
 drivers/staging/fsl_qbman/qman_config.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/staging/fsl_qbman/qman_config.c b/drivers/staging/fsl_qbman/qman_config.c
index 6a6225e..2ce3255 100644
--- a/drivers/staging/fsl_qbman/qman_config.c
+++ b/drivers/staging/fsl_qbman/qman_config.c
@@ -824,8 +824,8 @@ int qman_ceetm_set_prescaler(enum qm_dc_portal portal)
 	temp = 0x400000 * 100;
 	do_div(temp, CONFIG_QMAN_CEETM_UPDATE_PERIOD);
 	temp *= 10000000;
-	pres = (u16)(temp / qman_clk);
-
+	do_div(temp, qman_clk);
+	pres = (u16) temp;
 	qm_out(CEETM_CFG_IDX, portal);
 	qm_out(CEETM_CFG_PRES, pres);
 	return 0;
-- 
2.1.0.27.g96db324

